colossalai/zero/low_level/bookkeeping/gradient_store.py

```diff
@@ -107,16 +105,15 @@ def get_working_grad_by_param_id(self, param_id) -> Tensor:
         for group in self._grads_of_params.values():
             if param_id in group.keys():
                 return group[param_id][self._working_index]
-
-        raise KeyError(f"Working gradient for param_id {param_id} not found.")
+        return None
 
     def reset_grads_by_group_id(self, group_id: int):
         self._grads_of_params[group_id] = dict()
 
     def reset_all_gradients(self):
         self._grads_of_params = dict()
 
-    def get_param_id_for_grad(self, grad: Tensor) -> int:
+    def get_param_id_for_grad(self, grad: Tensor) -> Optional[int]:
```

Comment:
In which case is it None?

Reply:
This is for [get_param_id_for_grad](https://github.com/hpcaitech/ColossalAI/blob/5e551f875afbea0ce1f4ae3738ebc38d169ddecd/colossalai/zero/low_level/low_level_optim.py#L872).

Currently, grads are stored in different stores and I will need to check every store to see which store the grad belongs to (in this case `None` means not in the store, `int` means yes), a workaround is to maintain a grad-to-param mapping and update it for each backward.